# Use external HTSlib from EasyBuild environment. Based on the similar patch for v2.7.10b.
# See also PR https://github.com/alexdobin/STAR/pull/1586
# Author: maxim-masterov (SURF)
diff -Nru STAR-2.7.11a.orig/source/BAMfunctions.cpp STAR-2.7.11a/source/BAMfunctions.cpp
--- STAR-2.7.11a.orig/source/BAMfunctions.cpp	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/BAMfunctions.cpp	2023-08-17 15:40:33.432650000 +0200
@@ -1,5 +1,5 @@
 #include "BAMfunctions.h"
-#include "htslib/htslib/kstring.h"
+#include "htslib/kstring.h"
 
 
 string bam_cigarString (bam1_t *b) {//output CIGAR string
diff -Nru STAR-2.7.11a.orig/source/IncludeDefine.h STAR-2.7.11a/source/IncludeDefine.h
--- STAR-2.7.11a.orig/source/IncludeDefine.h	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/IncludeDefine.h	2023-08-17 15:41:18.348816000 +0200
@@ -30,8 +30,8 @@
 #define ERROR_OUT string ( __FILE__ ) +":"+ to_string ( (uint) __LINE__ ) +":"+ string ( __FUNCTION__ )
 
 //external libs
-#define SAMTOOLS_BGZF_H "htslib/htslib/bgzf.h"
-#define SAMTOOLS_SAM_H  "htslib/htslib/sam.h"
+#define SAMTOOLS_BGZF_H "htslib/bgzf.h"
+#define SAMTOOLS_SAM_H  "htslib/sam.h"
 
 using namespace std;
 
diff -Nru STAR-2.7.11a.orig/source/Makefile STAR-2.7.11a/source/Makefile
--- STAR-2.7.11a.orig/source/Makefile	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/Makefile	2023-08-17 15:42:02.606631000 +0200
@@ -12,8 +12,8 @@
 CXX ?= g++
 
 # pre-defined flags
-LDFLAGS_shared := -pthread -Lhtslib -Bstatic -lhts -Bdynamic -lz
-LDFLAGS_static := -static -static-libgcc -pthread -Lhtslib -lhts -lz
+LDFLAGS_shared := -pthread -L$(EBROOTHTSLIB)/lib -Bstatic -lhts -Bdynamic -lz
+LDFLAGS_static := -static -static-libgcc -pthread -L$(EBROOTHTSLIB)/lib -lhts -lz
 LDFLAGS_Mac :=-pthread -lz htslib/libhts.a
 LDFLAGS_Mac_static :=-pthread -lz -static-libgcc htslib/libhts.a
 LDFLAGS_gdb := $(LDFLAGS_shared)
@@ -131,7 +131,7 @@
 ifneq ($(MAKECMDGOALS),clean_solo)
 ifneq ($(MAKECMDGOALS),STARforMac)
 ifneq ($(MAKECMDGOALS),STARforMacGDB)
-Depend.list: $(SOURCES) parametersDefault.xxd htslib
+Depend.list: $(SOURCES) parametersDefault.xxd
 	echo $(SOURCES)
 	'rm' -f ./Depend.list
 	$(CXX) $(CXXFLAGS_common) -MM $^ >> Depend.list
diff -Nru STAR-2.7.11a.orig/source/STAR.cpp STAR-2.7.11a/source/STAR.cpp
--- STAR-2.7.11a.orig/source/STAR.cpp	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/STAR.cpp	2023-08-17 15:42:41.072327000 +0200
@@ -30,7 +30,7 @@
 
 #include "twoPassRunPass1.h"
 
-#include "htslib/htslib/sam.h"
+#include "htslib/sam.h"
 #include "parametersDefault.xxd"
 
 void usage(int usageType)
diff -Nru STAR-2.7.11a.orig/source/bamRemoveDuplicates.cpp STAR-2.7.11a/source/bamRemoveDuplicates.cpp
--- STAR-2.7.11a.orig/source/bamRemoveDuplicates.cpp	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/bamRemoveDuplicates.cpp	2023-08-17 15:40:50.759887000 +0200
@@ -1,7 +1,7 @@
 #include <unordered_map>
 #include "bamRemoveDuplicates.h"
 #include <iostream>
-#include "htslib/htslib/sam.h"
+#include "htslib/sam.h"
 #include "IncludeDefine.h"
 #include SAMTOOLS_BGZF_H
 #include "ErrorWarning.h"
diff -Nru STAR-2.7.11a.orig/source/bam_cat.c STAR-2.7.11a/source/bam_cat.c
--- STAR-2.7.11a.orig/source/bam_cat.c	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/bam_cat.c	2023-08-17 15:39:38.314318000 +0200
@@ -52,8 +52,8 @@
 #include <stdlib.h>
 #include <unistd.h>
 
-#include "htslib/htslib/bgzf.h"
-#include "htslib/htslib/sam.h"
+#include "htslib/bgzf.h"
+#include "htslib/sam.h"
 #include <cstring>
 
 #define BUF_SIZE 0x10000
diff -Nru STAR-2.7.11a.orig/source/bam_cat.h STAR-2.7.11a/source/bam_cat.h
--- STAR-2.7.11a.orig/source/bam_cat.h	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/bam_cat.h	2023-08-17 15:39:59.175559315 +0200
@@ -1,7 +1,7 @@
 #ifndef CODE_bam_cat
 #define CODE_bam_cat
 
-#include "htslib/htslib/sam.h"
+#include "htslib/sam.h"
 
 int bam_cat(int nfn, char * const *fn, const bam_hdr_t *h, const char* outbam);
 
diff -Nru STAR-2.7.11a.orig/source/signalFromBAM.h STAR-2.7.11a/source/signalFromBAM.h
--- STAR-2.7.11a.orig/source/signalFromBAM.h	2023-08-15 21:21:58.000000000 +0200
+++ STAR-2.7.11a/source/signalFromBAM.h	2023-08-17 15:42:27.321267434 +0200
@@ -1,6 +1,6 @@
 #ifndef CODE_signalFromBAM
 #define CODE_signalFromBAM
-#include "htslib/htslib/sam.h"
+#include "htslib/sam.h"
 #include  <fstream>
 #include <string>
 #include "Stats.h"
